presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -282,6 +284,18 @@ private void chooseDistribution(PartitioningHandle distribution)
                     partitioningHandle);
         }
 
+        private boolean isCompatibleSystemPartitioning(PartitioningHandle distribution)
+        {
+            ConnectorPartitioningHandle currentHandle = partitioningHandle.get().getConnectorHandle();
+            ConnectorPartitioningHandle distributionHandle = distribution.getConnectorHandle();
+            if ((currentHandle instanceof SystemPartitioningHandle) &&
+                    (distributionHandle instanceof SystemPartitioningHandle)) {
+                return ((SystemPartitioningHandle) currentHandle).getPartitioning() ==
```

Comment:
It seems to me like they are. Look on `com.facebook.presto.sql.planner.NodePartitioningManager#getNodePartitioningMap`.  In first lines there is this if condition:
```
        if (partitioningHandle.getConnectorHandle() instanceof SystemPartitioningHandle) {
            return ((SystemPartitioningHandle) partitioningHandle.getConnectorHandle()).getNodePartitionMap(session, nodeScheduler);
        }
```
and in `com.facebook.presto.sql.planner.SystemPartitioningHandle#getNodePartitionMap`:
```
        NodeSelector nodeSelector = nodeScheduler.createNodeSelector(null);
        List<Node> nodes;
        if (partitioning == SystemPartitioning.COORDINATOR_ONLY) {
            nodes = ImmutableList.of(nodeSelector.selectCurrentNode());
        }
        else if (partitioning == SystemPartitioning.SINGLE) {
            nodes = nodeSelector.selectRandomNodes(1);
        }
        else if (partitioning == SystemPartitioning.FIXED) {
            nodes = nodeSelector.selectRandomNodes(getHashPartitionCount(session));
        }
        else {
            throw new IllegalArgumentException("Unsupported plan distribution " + partitioning);
        }
```
 those seems to me like the only relevant places, that implement this "compatibility" (or lack of it). As long as `partitioning` are the same, there are no difference in behavior, especially for all possible types of `FIXED` the same code should execute everywhere. I was even playing safe here, because it looks like even both `SINGLE` and `FIXED` could collapse to `SINGLE`.

I don't like that this code in `PlanFragmenter` depends so much on implementation of scheduler classes, but without some heavy refactor I am not sure if we can fix that. Especially in "release blocker" mode.